File: VB/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Xpf.Reports.UserDesigner
Imports DevExpress.XtraReports.UI
Imports System
Imports System.Linq
Imports System.Windows
Imports System.Windows.Resources

Namespace ManipulateReportDesignerDocuments_MVVM
	Public Class MainViewModel
		Inherits ViewModelBase

		Public Property DefaultReport() As XtraReport

		Private ReadOnly Property ReportDesignerAPI() As IReportDesignerAPIService
			Get
				Return ServiceContainer.GetService(Of IReportDesignerAPIService)()
			End Get
		End Property

		Protected Overrides Sub OnInitializeInRuntime()
			MyBase.OnInitializeInRuntime()
			DefaultReport = New XtraReport()
		End Sub

		<Command>
		Public Sub NewReport()
			ReportDesignerAPI.NewReport()
		End Sub
		Public Function CanNewReport() As Boolean
			Return ReportDesignerAPI IsNot Nothing
		End Function

		<Command>
		Public Sub OpenReport()
			ReportDesignerAPI.Open()
		End Sub
		Public Function CanOpenReport() As Boolean
			Return ReportDesignerAPI IsNot Nothing
		End Function

		<Command>
		Public Sub OpenReportFromStream()
			Dim stream = Application.GetResourceStream(New Uri("Resources/Invoice.repx", UriKind.Relative))?.Stream
			If stream IsNot Nothing Then
				Using stream
					ReportDesignerAPI.Open(stream)
				End Using
			End If
		End Sub
		Public Function CanOpenReportFromStream() As Boolean
			Return ReportDesignerAPI IsNot Nothing
		End Function

		<Command>
		Public Sub CloseActiveDocument()
			ReportDesignerAPI.ActiveDocument.Close(True)
		End Sub
		Public Function CanCloseActiveDocument() As Boolean
			Return ReportDesignerAPI?.ActiveDocument IsNot Nothing
		End Function

		<Command>
		Public Sub ActivateFirstDocumentTab()
			Dim firstTab As ReportDesignerDocument = ReportDesignerAPI.Documents.First()
			firstTab.Activate()
		End Sub
		Public Function CanActivateFirstDocumentTab() As Boolean
			Return If(ReportDesignerAPI?.Documents.Any(), False)
		End Function

		<Command>
		Public Sub SwitchActiveDocumentView()
			Dim activeDocument As ReportDesignerDocument = ReportDesignerAPI.ActiveDocument
			activeDocument.ViewKind = If(activeDocument.ViewKind = ReportDesignerDocumentViewKind.Designer, ReportDesignerDocumentViewKind.Preview, ReportDesignerDocumentViewKind.Designer)
		End Sub
		Public Function CanSwitchActiveDocumentView() As Boolean
			Return ReportDesignerAPI?.ActiveDocument IsNot Nothing
		End Function
	End Class
End Namespace

File: CS/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Xpf.Reports.UserDesigner;
using DevExpress.XtraReports.UI;
using System;
using System.Linq;
using System.Windows;
using System.Windows.Resources;

namespace ManipulateReportDesignerDocuments_MVVM {
    public class MainViewModel : ViewModelBase {
        public XtraReport DefaultReport { get; set; }

        IReportDesignerAPIService ReportDesignerAPI => ServiceContainer.GetService<IReportDesignerAPIService>();

        protected override void OnInitializeInRuntime() {
            base.OnInitializeInRuntime();
            DefaultReport = new XtraReport();
        }

        [Command]
        public void NewReport() {
            ReportDesignerAPI.NewReport();
        }
        public bool CanNewReport() => ReportDesignerAPI != null;

        [Command]
        public void OpenReport() {
            ReportDesignerAPI.Open();
        }
        public bool CanOpenReport() => ReportDesignerAPI != null;

        [Command]
        public void OpenReportFromStream() {
            var stream = Application.GetResourceStream(new Uri("Resources/Invoice.repx", UriKind.Relative))?.Stream;
            if(stream != null) {
                using(stream) {
                    ReportDesignerAPI.Open(stream);
                }
            }
        }
        public bool CanOpenReportFromStream() => ReportDesignerAPI != null;

        [Command]
        public void CloseActiveDocument() {
            ReportDesignerAPI.ActiveDocument.Close(true);
        }
        public bool CanCloseActiveDocument() => ReportDesignerAPI?.ActiveDocument != null;

        [Command]
        public void ActivateFirstDocumentTab() {
            ReportDesignerDocument firstTab = ReportDesignerAPI.Documents.First();
            firstTab.Activate();
        }
        public bool CanActivateFirstDocumentTab() => ReportDesignerAPI?.Documents.Any() ?? false;

        [Command]
        public void SwitchActiveDocumentView() {
            ReportDesignerDocument activeDocument = ReportDesignerAPI.ActiveDocument;
            activeDocument.ViewKind = activeDocument.ViewKind == ReportDesignerDocumentViewKind.Designer
                ? ReportDesignerDocumentViewKind.Preview
                : ReportDesignerDocumentViewKind.Designer;
        }
        public bool CanSwitchActiveDocumentView() => ReportDesignerAPI?.ActiveDocument != null;
    }
}
